ANTLR not found while building

In GitLab by @EricMier on Mar 9, 2022, 16:50

On some systems the cmake config for ANTLR can't be found, resulting in following error message:

    CMake Error at src/parser/CMakeLists.txt:32 (find_package):
      By not providing "Findantlr4-generator.cmake" in CMAKE_MODULE_PATH this
      project has asked CMake to find a package configuration file provided by
      "antlr4-generator", but CMake did not find one.

      Could not find a package configuration file provided by "antlr4-generator"
      with any of the following names:

        antlr4-generatorConfig.cmake
        antlr4-generator-config.cmake

      Add the installation prefix of "antlr4-generator" to CMAKE_PREFIX_PATH or
      set "antlr4-generator_DIR" to a directory containing one of the above
      files.  If "antlr4-generator" provides a separate development package or
      SDK, be sure it has been installed.

This is due to incorrect path generation in [src/parser/CMakeLists.txt#L28](https://gitlab.know-center.tugraz.at/daphne/prototype/-/blob/master/src/parser/CMakeLists.txt#L28), where on some systems ${CMAKE_INSTALL_LIBDIR} is resolved to "lib/x86_64-linux-gnu" resulting in:

    antlr4-generator_DIR = <project>/thirdparty/antlr/antlr4-cpp-runtime-4.9.2-source/run/usr/local/lib/x86_64-linux-gnu/cmake/antlr4-generator

whereas the required cmake config files reside in:

    <project>/thirdparty/antlr/antlr4-cpp-runtime-4.9.2-source/run/usr/local/lib/cmake/antlr4-generator

Same goes for antlr4-runtime.

--- a/build.sh
+++ b/build.sh
@@ -149,6 +149,9 @@ then
     mkdir --parents $antlrCppRuntimeDirName
     unzip $antlrCppRuntimeZipName -d $antlrCppRuntimeDirName
     cd $antlrCppRuntimeDirName
+    # Github disabled the unauthenticated git:// protocol, patch antlr4 to use https://
+    # until we upgrade to antlr4-4.9.3+
+    sed -i 's#git://github.com#https://github.com#' runtime/CMakeLists.txt
     rm -rf ./build
     mkdir -p build
     mkdir -p run

